### ISSUE TITLE
Change to uint16_t to support messages > 255 bytes

### ISSUE THREAD

**Issue by jwende** [adafruit/Adafruit_MQTT_Library#104]
https://github.com/adafruit/Adafruit_MQTT_Library/blob/974f4b8713a83d01b040a0d23b798d3f2978d990/Adafruit_MQTT.cpp#L228

Cheers
Joerg

**Comment by rareearthmagnet** [comment]
Jeorg, do if this is the only thing in the code that needs to be changed to support larger messages? I am still having problems sending messages > 100 bytes.... 

Thanks.

**Comment by jwende** [comment]
that is correct (and there might be another issue with the 23 byte limit of the device identifier ... this had been removed in MQTT 3.1.1)

**Comment by jjunimd** [comment]
By "device identifier" do you mean the server?  The feed name?  Since these are URLs, what portion of them counts toward the 23 byte limit?

**Comment by jwende** [comment]
sorry - Client Identifier 
The Client Identifier (ClientId) identifies the Client to the Server. Server may allow client identifies larger than 23 bytes
(http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718028)

**Comment by jjunimd** [comment]
Thanks!
